includes/embeds/class-amp-base-embed-handler.php

```diff
@@ -109,4 +109,43 @@ function ( $attr_name ) {
 		}
 		return wp_array_slice_assoc( $matches, $attribute_names );
 	}
+
+	/**
+	 * Get all child elements of the specified element.
+	 *
+	 * @since 2.0.6
+	 *
+	 * @param DOMElement $node Element.
+	 * @return DOMElement[] Array of child elements for specified element.
+	 */
+	protected function get_child_elements( DOMElement $node ) {
+		return array_filter(
+			iterator_to_array( $node->childNodes ),
+			static function ( DOMNode $child ) {
+				return $child instanceof DOMElement;
+			}
+		);
+	}
+
+	/**
+	 * Replace the node's parent with itself if the parent is a <p> tag, has no attributes and has no other children.
+	 * This usually happens while `wpautop()` processes the element.
+	 *
+	 * @since 2.0.6
+	 *
+	 * @param DOMElement $node Node.
+	 */
+	protected function unwrap_p_element( DOMElement $node ) {
+		$parent_node = $node->parentNode;
+		while ( $parent_node && ! ( $parent_node instanceof DOMElement ) ) {
```

Comment:
When will `$parent_node` not be an instance of `DOMElement`?

Reply:
It was added as a sanity check since it is technically possible for `$parent_node` to be a `DOMNode`, although in this circumstance that does seem unlikely.

I've removed the while loop in https://github.com/ampproject/amp-wp/pull/5548/commits/16b39838806fe626d73126862653adccf04ad6a3.